h2o-automl/src/main/java/ai/h2o/automl/TrainingStepsExecutor.java

```diff
@@ -0,0 +1,180 @@
+package ai.h2o.automl;
+
+import ai.h2o.automl.EventLogEntry.Stage;
+import ai.h2o.automl.WorkAllocations.JobType;
+import ai.h2o.automl.WorkAllocations.Work;
+import hex.Model;
+import hex.grid.Grid;
+import water.Iced;
+import water.Job;
+import water.Key;
+import water.util.Countdown;
+import water.util.Log;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+
+/**
+ * Class responsible for starting all the {@link TrainingStep}s and monitoring their associated {@link Job},
+ * i.e. polling jobs and adding their result model(s) to the {@link Leaderboard}.
+ */
+class TrainingStepsExecutor extends Iced<TrainingStepsExecutor> {
+
+    EventLog _eventLog;
+    Leaderboard _leaderboard;
```

Comment:
this will serialize the leaderboard, but doesn't leaderboard have its own update mechanism? should we just remember the leaderboard key here?

Reply:
yes, we should, will change that.